### PR TITLE
Add Rancher air gap installation code flow

### PR DESCRIPTION
**Pull request by yangchiu**
Add Rancher air gap installation code flow

for https://github.com/longhorn/longhorn/issues/3926

Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

**Comment by innobead** [comment]
This is great. We can base on this to have Rancher integration testing for the Longhorn chart in general.

File: test_framework/terraform/aws/centos/main.tf [hunk 1]
```diff
@@ -268,4 +284,95 @@ resource "aws_ebs_volume" "lh_aws_hdd_volume" {
   tags = {
     Name = "lh-aws-hdd-volume-${random_string.random_suffix.id}-${count.index}"
   }
+}
+
+# Create load balancer for rancher
+resource "aws_lb_target_group" "lh_aws_lb_tg_443" {
```
Comment (innobead) [inline test_framework/terraform/aws/centos/main.tf:290]: For each new resource, need to add a tag for our usage identity.

File: test_framework/scripts/longhorn-setup.sh [hunk 2]
```diff
@@ -54,6 +54,48 @@ install_csi_snapshotter_crds(){
...
+}
+
+
+install_longhorn_by_rancher() {
+  terraform -chdir="${TF_VAR_tf_workspace}/scripts/rancher/terraform" init
+  terraform -chdir="${TF_VAR_tf_workspace}/scripts/rancher/terraform" apply \
+            -var="api_url=https://${RANCHER_HOSTNAME}" \
+            -var="access_key=${RANCHER_ACCESS_KEY}" \
+            -var="secret_key=${RANCHER_SECRET_KEY}" \
+            -var="longhorn_git_repo=${LONGHORN_RANCHER_CHART_REPO}" \
```
Comment (innobead) [inline test_framework/scripts/longhorn-setup.sh:89]: Suggest renaming to `rancher_chart_git_repo` and the other is `rancher_chart_git_branch`

File: test_framework/scripts/rancher/terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  git_branch = var.longhorn_git_branch
+}
+
+resource "rancher2_app_v2" "longhorn_app" {
+  cluster_id = "local"
+  name = "longhorn-app"
+  namespace = "longhorn-system"
+  repo_name = "longhorn-repo"
+  chart_name = "longhorn"
+  #chart_version = "100.1.2+up1.2.4" # if chart_version is not specified, the latest version would be installed
```
Comment (innobead) [inline test_framework/scripts/rancher/terraform/main.tf:30]: This is good. Probably we should keep this, then we can test different versions if needed in other cases.

File: test_framework/terraform/aws/centos/main.tf [hunk 4]
```diff
@@ -268,4 +284,95 @@ resource "aws_ebs_volume" "lh_aws_hdd_volume" {
...
+
+  count = var.create_load_balancer ? 1 : 0
+
+  depends_on = [
+    aws_lb.lh_aws_lb,
+    aws_lb_target_group.lh_aws_lb_tg_80
+  ]
+
+  load_balancer_arn = aws_lb.lh_aws_lb[0].arn
+  port              = "80"
```
Comment (innobead) [inline test_framework/terraform/aws/centos/main.tf:370]: Do we need 80 port LB? In main.py, it just used https to communicate with the Rancher server.

**Review by innobead — COMMENTED** [review]
In general, LGTM. Just a few comments requiring to resolve.

File: test_framework/terraform/aws/centos/main.tf (hunk 1 above)
Comment (yangchiu) [reply test_framework/terraform/aws/centos/main.tf:290]: Modified.

File: test_framework/scripts/longhorn-setup.sh (hunk 2 above)
Comment (yangchiu) [reply test_framework/scripts/longhorn-setup.sh]: Modified.

File: test_framework/terraform/aws/centos/main.tf (hunk 4 above)
Comment (yangchiu) [reply test_framework/terraform/aws/centos/main.tf]: Modified.

File: test_framework/scripts/rancher/terraform/main.tf (hunk 3 above)
Comment (yangchiu) [reply test_framework/scripts/rancher/terraform/main.tf]: Modified.

Comment (innobead) [reply test_framework/scripts/rancher/terraform/main.tf]: 👍

File: test_framework/terraform/aws/centos/main.tf [hunk 5]
```diff
@@ -268,4 +284,70 @@ resource "aws_ebs_volume" "lh_aws_hdd_volume" {
...
+
+  count = var.create_load_balancer ? 1 : 0
+
+  name     = "lh-aws-lb-tg-443-${random_string.random_suffix.id}"
+  port     = 443
+  protocol = "TCP"
+  vpc_id   = aws_vpc.lh_aws_vpc.id
+  health_check {
+    protocol = "TCP"
+    port = "80"
```
Comment (innobead) [inline test_framework/terraform/aws/centos/main.tf:300]: Why is this 80? This should be the port opened on each worker node and able to forward to Rancher.

Comment (yangchiu) [reply test_framework/terraform/aws/centos/main.tf:300]: It follows the document: https://rancher.com/docs/rancher/v2.6/en/installation/resources/k8s-tutorials/infrastructure-tutorials/nlb/
But other experiments can be done to check the effect on modification of this value.

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@yangchiu We can backport this to 1.3 and 1.2 as well for the following patch release.

**Pull request merged**